lib/internal/fs/streams.js

```diff
@@ -224,6 +233,11 @@ ReadStream.prototype._destroy = function(err, cb) {
     return;
   }
 
+  if (this.isPerformingIO) {
+    this.once(kIoDone, closeFsStream.bind(null, this, cb, err));
+    return;
+  }
+
   closeFsStream(this, cb, err);
   this.fd = null;
```

Comment:
I think some tests expect `fd === null` directly after `destroy()`, changing this might make it semver-major.

Reply:
I don’t think it needs to be.

Reply:
I think we should probably set `this.fd = null` inside `closeFsStream`. Otherwise it won't be set to null in the new branch.

Reply:
Sure, done